electron/legendary/games.ts

```diff
@@ -216,6 +216,61 @@ class LegendaryGame extends Game {
     return newInstallPath
   }
 
+  // used when downloading games, store the download size read from Legendary's output
+  currentDownloadSize = 0
+
+  public onInstallOrUpdateOutput(
+    action: 'installing' | 'updating',
+    totalDownloadSize: number,
+    data: string
+  ) {
+    const downloadSizeMatch = data.match(/Download size: ([\d.]+) MiB/)
+
+    // store the download size, needed for correct calculation
+    // when cancel/resume downloads
+    if (downloadSizeMatch) {
+      this.currentDownloadSize = parseFloat(downloadSizeMatch[1])
+    }
+
+    // parse log for game download progress
+    const etaMatch = data.match(/ETA: (\d\d:\d\d:\d\d)/m)
+    const bytesMatch = data.match(/Downloaded: (\S+.) MiB/m)
+    if (etaMatch && bytesMatch) {
+      const eta = etaMatch[1]
+      const bytes = bytesMatch[1]
+
+      // original is in bytes, convert to MiB with 2 decimals
+      totalDownloadSize =
+        Math.round((totalDownloadSize / 1024 / 1024) * 100) / 100
```

Comment:
why dont you use Pretty-Bytes here? its the library we are already using for Byts convertion.

Reply:
I noticed pretty bytes shows a different value than Legendary

I was testing with Absolute Drift for example and in the install dialog it shows 158 (seems to be dividing by 1000 using MB) but then Legendary reports 150.71 (seems to be dividing by 1024 using MiB) so the calculations were off (maybe I'm just missing some option for the prettyBytes function to set the unit)

Also, pretty bytes returns a string and then I would have to convert it back to a number for the calculations

one comemnt though, maybe we can remove the pretty bytes dependency and move this code to a helper? this type of calculation seems simple enough to not need an external dependency since we don't need that much flexibility, should be a simple refactor

Reply:
Yes, could be better then. So we could use the same approach for all and get rid of a library.

Reply:
I can do the refactor in another PR, it's used in many places and it's kinda unrelated to this change, I'll keep a note and refactor that in another moment to keep this PR simpler